experimental/dds/stack/src/sharedStack.ts

```diff
@@ -0,0 +1,162 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IEvent } from "@fluidframework/common-definitions";
+import { IsoBuffer } from "@fluidframework/common-utils";
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { IChannel, IChannelAttributes, IChannelFactory, IChannelServices, IChannelStorageService, IFluidDataStoreRuntime } from "@fluidframework/datastore-definitions";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+import { ITelemetryContext, ISummaryTreeWithStats, IGarbageCollectionData } from "@fluidframework/runtime-definitions";
+import { SummaryTreeBuilder } from "@fluidframework/runtime-utils";
+import { IFluidSerializer, SharedObject } from "@fluidframework/shared-object-base";
+
+/**
+ * Events useful for testing and debugging a {@link SharedStack}
+ */
+export interface DiagnosticEvents extends IEvent {
+    (event: "downloadedBlob", handler: () => void);
+    (event: "uploadedBlob", handler: () => void);
+    (event: "processedOp", handler: () => void);
+}
+
+/**
+ * A stack DDS which stores every element in its own attachment blob.
+ */
+export class SharedStack<T = unknown> extends SharedObject<DiagnosticEvents> {
```

Comment:
T could be constrained to Serializable<T>.

Reply:
I'm not sure how to use that type here. `Serializable` === `any`. `Serializable<T>` will be `never` if `T` is not serializable, but that still doesn't give us a good way to constrain `T`.